### PR TITLE
INGK-467 Add support to socketcan (Linux)

### PR DESCRIPTION
**Pull request by pzinemanas**
### Description

In this PR the support to socketcan device is added. So the canopen communication can be used in Linux platforms.

Fixes # INGK-467

### Type of change

Please add a description and delete options that are not relevant.

- [x] Add support to socketcan.
- [x] Fix import errors in linux.

### Tests
- [x] Test connection and SDOs.
- [x] Test load FW.
- [x] Run unit tests in a Linux machine.

Please describe the tests that you ran to verify your changes if it applies. 

### Documentation

Please update the documentation.

- [x] Update docstrings of every function, method or class that change.
- [x] Build documentation locally to verify changes.
- [x] Add the changes at the `[Unreleased]` section of the [CHANGELOG](CHANGELOG.md).

### Code formatting

- [x] Use black package to format the code: `black -l 100 ingenialink tests`. It is recommended to configure the code editor to automatically format the code using black with a max length line of 100.

### CODE REVIEW

File: ingenialink/canopen/network.py [hunk 1]
```diff
@@ -354,6 +362,8 @@ def _setup_connection(self) -> None:
         """Creates a network interface object establishing an empty connection
         with all the network attributes already specified."""
         if self._connection is None:
+            if not RUNNING_ON_WINDOWS and self.__device != CAN_DEVICE.SOCKETCAN.value:
```
Comment (MartinALL91) [inline ingenialink/canopen/network.py:365]: I would move this check to the constructor.

File: ingenialink/canopen/network.py [hunk 2]
```diff
@@ -8,10 +8,16 @@
 from time import sleep
 from typing import Any, Callable, Dict, List, Optional, Tuple, Union
 
+RUNNING_ON_WINDOWS = os.name == "nt"
```
Comment (MartinALL91) [inline ingenialink/canopen/network.py:11]: We could use the [system](https://docs.python.org/3/library/platform.html#platform.system) method to check the OS.

File: ingenialink/canopen/network.py [hunk 3]
```diff
@@ -631,11 +641,16 @@ def __program_control_to_stop(
         logger.info("Flashing firmware")
         with contextlib.suppress(ILError):
             servo.write(PROG_STAT_1, PROG_CTRL_STATE_STOP, subnode=0)
-        try:
+        if RUNNING_ON_WINDOWS:
```
Comment (MartinALL91) [inline ingenialink/canopen/network.py:644]: I don't think this is necessary.

**Review by MartinALL91 — CHANGES_REQUESTED** [review]

File: ingenialink/canopen/network.py (hunk 1 above)
Comment (MartinALL91) [reply ingenialink/canopen/network.py:365]: We should check if the OS is linux.

File: ingenialink/canopen/network.py (hunk 3 above)
Comment (pzinemanas) [reply ingenialink/canopen/network.py:644]: The thing is that the VCIError would not be defined when running on windows.

File: ingenialink/canopen/network.py (hunk 2 above)
Comment (pzinemanas) [reply ingenialink/canopen/network.py]: Done [9d6f708](https://github.com/ingeniamc/ingenialink-python/pull/390/commits/9d6f708707fbcf386670c5534ef19ed948f7e0bb)

File: ingenialink/canopen/network.py (hunk 1 above)
Comment (pzinemanas) [reply ingenialink/canopen/network.py]: Done [9d6f708](https://github.com/ingeniamc/ingenialink-python/pull/390/commits/9d6f708707fbcf386670c5534ef19ed948f7e0bb)

File: ingenialink/canopen/network.py [hunk 4]
```diff
@@ -185,6 +194,8 @@ def __init__(
         baudrate: CAN_BAUDRATE = CAN_BAUDRATE.Baudrate_1M,
     ):
         super(CanopenNetwork, self).__init__()
+        if not platform.system() == "Linux" and device != CAN_DEVICE.SOCKETCAN.value:
+            raise ILError("In Linux machines, only the SOCKETCAN device can be used.")
```
Comment (GullinBustin) [inline ingenialink/canopen/network.py:198]: Why? Technically the other drivers can be used too.

**Review by GullinBustin — CHANGES_REQUESTED** [review]

File: ingenialink/canopen/network.py (hunk 4 above)
Comment (pzinemanas) [reply ingenialink/canopen/network.py:198]: That's right. I'll remove this error.

**Review by MartinALL91 — APPROVED** [review]

**Pull request merged**